docs/source/mistral_yaql.rst

```diff
@@ -0,0 +1,201 @@
+        type: direct
+        input:
+            - ip
+            - port
+        tasks:
+            task1:
+                action: examples.call-api
+                input:
+                    endpoint: https://<% $.ip %>:<% $.port>/api
+
+Certain statements in Mistral such as on-success and on-error can evaluate boolean logic. The 
+``on-condition`` related statements are used for transition from one task to another. If a 
+boolean logic is defined with these statements, it can be used to evaluate whether the transition
+should continue or not. Complex boolean logic using a combination of ``not``, ``and``, ``or``, and
+parentheses is possible. Take the following workflow as an example, execution of certain branch 
+in the workflow depends on the value of ``$.path``. If ``$.path = a``, then task ``a`` is executed. 
+If ``$.path = b``, then task ``b``. Finally task ``c`` is executed if neither.
+
+.. literalinclude:: /../../contrib/examples/actions/workflows/mistral-branching.yaml
+
+Dictionaries
+++++++++++++
+To create a dictionary, use the ``dict`` function. For example, ``<% dict(a=>123, b=>true) %>``
+returns ``{'a': 123, 'b': True}``. Let's say this dictionary is published to the context as
+``dict1``, the keys function returns ``<% $.dict1.keys() %>`` returns ``['a', 'b']`` and
+``<% $.dict1.values() %>`` returns the values ``[123, true]``. Concatenating dictionaries
+can be done as ``<% dict(a=>123, b=>true) + dict(c=>xyz) %>`` which returns
+``{'a': 123, 'b': True, 'c': 'xyz'}``. Specific key-value pair can be accessed by key name
+such as ``<% $.dict1.get(b) %>`` which returns ``True``. Given the alternative
+``<% $.dict1.get(b, false) %>`` and lets say the key ``b`` does not exist, then ``False``
+will be returned by default.
+
+Lists
++++++
+To create a list, use the ``list`` functions. For example, ``<% list(1, 2, 3) %>`` returns 
+``[1, 2, 3]`` and ``<% list(abc, def) %>`` returns ``['abc', 'def']``. List concatenation  
+can be done as ``<% list(abc, def) + list(ijk, xyz) %>`` which returns 
+``['abc', 'def', 'ijk', 'xyz']``. Let's say this list is published to the context as ``list1``,
+items can also be access via index such as ``<% $.list1[0] %>`` which returns ``abc``.
+
+Queries
++++++++
+Let's take the following context as an example.
+
+.. code-block:: json
+
+    {
+        'vms': [
+            {
+                'name': 'vmweb1',
+                'region': 'us-east',
+                'role': 'web'
+            },
+            {
+                'name': 'vmdb1',
+                'region': 'us-east',
+                'role': 'db'
+            },
+            {
+                'name': 'vmweb2',
+                'region': 'us-west',
+                'role': 'web'
+            },
+            {
+                'name': 'vmdb2',
+                'region': 'us-west',
+                'role': 'db'
+            }
+        ]
+    }
+
+The YAQL expression ``<% $.vms.select($.name) %>`` returns only the list of VM names.
```

Comment:
If possible can you paste the full output for each of the expressions? Having the exact output would help comprehension. Not sure if that would be too tedious.